src/twisted/internet/defer.py

```diff
@@ -160,6 +167,87 @@ def maybeDeferred(f, *args, **kw):
 
 
 
+def extractCurrentResult(deferred, raises=True, passthrough=False):
+    """
+    Extract the current result from a L{Deferred} that has already fired all of
+    its callbacks.
+
+    Take a deferred that has already fired all of its callbacks and extract and
+    return the result of the L{Deferred}. This process of extracting the result
+    will, by default, consume the result, error or not unless the
```

Comment:
I think ` ... consume the result - error or not - unless the ...`.  Certainly not just a single comma before the "error or not", though.  The aside is just "error or not" and it should be delimited on both ends by some punctuation.  This could still be commas but there's already a lot of commas here so I think dashes would be less confusing.